### ISSUE TITLE
Autocompletion doesn't work with decorators

### ISSUE THREAD

**Issue by 19majkel94** [buehler/typescript-hero#111]
At the time when we had Angular2 and other Typescript frameworks which use `@decorators`, it would be nice to have an working autocompletion on decorators like when we start to type `@Comp` it should show as import suggestion `Component` from `@angular/core` 😉

**Comment by buehler** [comment]
Hey @19majkel94 
Actually, it does work (with giuseppe at least :-) )
it seems I need to investigate angular a little bit more. They also produce a max stack size because the have so many export :-(

**Comment by 19majkel94** [comment]
Hi @buehler!
Angular is just an example of popular decorator use, the autocompletion also doesn't work with [TypeORM](/typeorm/typeorm):

![2016-10-27 21_12_17-startowa](https://cloud.githubusercontent.com/assets/10618781/19781626/671061fe-9c8a-11e6-95d2-a73779ae2e23.png)

```typescript
export declare function Table(name?: string, options?: TableOptions): (target: Function) => void;
```

It doesn't trigger suggestion box when `@` typed, `CTRL+SPACE` open its but there's no functions (like decorators functions), only classes.

**Comment by buehler** [comment]
@19majkel94 is this still an issue?

**Comment by 19majkel94** [comment]
I don't know, I've switched to other extension a long time ago 😞
